compiler/src/iree/compiler/Codegen/Common/GPU/GPUPromoteMatmulOperands.cpp

```diff
@@ -53,9 +53,17 @@ void promoteOperand(OpBuilder &builder, Operation *op, unsigned index) {
         return;
       }
     }
-    setLoweringConfig(producer, IREE::GPU::DerivedThreadConfigAttr::get(
-                                    builder.getContext()));
-    return;
+
+    bool promoteProducer = true;
+    if (isa<tensor::PadOp>(producer)) {
+      promoteProducer = false;
+    }
+
+    if (promoteProducer) {
```

Comment:
nit: Just `!isa<tensor::PadOp>(producer)` or `isa<linalg::LinalgOp, LinalgExt::Im2ColOp>(producer)` since `promoteProducer` is only used once.

Reply:
made it isa<linalg::LinalgOp, LinalgExt::Im2ColOp>